lib/ansible/modules/files/file.py

```diff
@@ -388,6 +388,9 @@ def main():
         elif prev_state in ('file', 'hard') and not force:
             module.fail_json(path=path, msg='refusing to convert between %s and %s for %s' % (prev_state, state, path))
 
+        if state == 'hard':
+            prev_state = 'hard'
```

Comment:
this will break other pathways as you are 'lying' to the code instead of having the code handle this particular combination correctly.